### PR TITLE
Feature/commissions/api support and tests

### PR DESCRIPTION
**Pull request by chdefrene**

### CODE REVIEW

**Comment by trmdy** [comment]
I am not sure I am in favour of adding serpy as a dependency, even though it's supposed to be a lot faster. For now I'd like us for simplicity to stick with plain DRF-ModelSerializers, which makes our life as easy as possible. 

We can have a discussion regarding what type of viewsets to add, but as the API is of secondary priority, I think I will advice everyone to just add a default ModelViewSet.

Also see the comments in #15.

**Comment by chdefrene** [comment]
I agree with your serpy argument, and have removed the dependency.

I don't agree with you notion of defaulting to ModelViewSets versus generics. There is not much difference to the programmer between these two methods, and personally I feel generics are a bit easier to understand, especially for programmers that are new to REST. But let's discuss this in person the next time you're around 🙂

**Comment by trmdy** [comment]
In the end I will let you decide on this. I'll just post my thoughts below, and you can take them into account.

My reasoning for preferring (Model)ViewSets, at least for now, is:

* It requires minimal coding to get a functioning CRUD API up and running.
* We will mostly adhere to a very neatly organized CRUD API, which means most code extensions we do will be of the middleware sort.
* It is very copy-pastable, which might make the life our developers (who will mostly not concern themselves with the API) easier.

I agree that to some degree Generics might be easier to digest for the inexperienced, as it is more explicit. However, it undeniably leads to *more* code unless one concerns oneself with *most* of the specific implementations. For instance, a more or less semantic equivalent of the above implementation would be less than 10 lines of code.

Consequently, I feel the burden is "on the generics" to argue why we *need* the extra 70 lines of code to get roughly the same level of functionality. 

Finally; the less code we have to write ourselves, the less chance we get to perform errors. We also would not have to write as many tests (as DRF's test coverage implicitly would cover much of our own use).

Note that if we go for the Generics version, might it perhaps be best to have one or two developers who are put on API-programming-duty? While the rest concern themselves with other parts of the backend-programming.

**Comment by trmdy** [comment]
Also, the idea is that the API lies on the `/api` endpoint, and that we have other template-related CRUD (functional) views which lie on the /internal/<app_name> views. The latter type of view is the primary views we are going to write for each model, that are interacted with by the webapp. 

The DRF API was simply suggested as a JSON-REST-API supplement we could very rapidly prototype (in part due to the ModelViewSets).

File: commissions/urls.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django.conf.urls import url
+
+from commissions import views
+
+urlpatterns = [
+    url(r'^$', views.ListCommissionsView.as_view(), name='list-commissions'),
```
Comment (trmdy) [inline commissions/urls.py]: These urls should as indicated in the PR-comment lie on the `/api` url.

File: commissions/views.py [hunk 2]
```diff
@@ -1,3 +1,83 @@
...
+    """
+    Create a new commission in the database.
+    """
+    model = Commission
+    permission_classes = [IsAuthenticated]
+    serializer_class = CreateCommissionSerializer
+    queryset = Commission.objects.all()
+
+    def post(self, request, *args, **kwargs):
+        if 'name' not in request.data:
```
Comment (trmdy) [inline commissions/views.py]: I feel perhaps these two if statements should be handled by uniqueness/blankness in the Model (and then automatically by the Serializer)?

File: commissions/urls.py (hunk 1 above)
Comment (chdefrene) [reply commissions/urls.py]: Agree

File: commissions/views.py (hunk 2 above)
Comment (chdefrene) [reply commissions/views.py]: You're (almost) right! This is handled by the serializer if you add a uniqueness validator to the serializer field.

**Review by trmdy — APPROVED** [review]
Looks good!

**Pull request merged**